src/main/java/com/musala/atmosphere/agent/util/FtpFileTransferService.java

```diff
@@ -0,0 +1,96 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
+
+import com.musala.atmosphere.agent.devicewrapper.AbstractWrapDevice;
+
+/**
+ * A service responsible for all file transfers to the FTP server. The service will add all finished screen records to a
+ * persistent queue(a text file with list of queued screen record names) and will trigger a transfer if the FTP
+ * connection manager is available for a new job.
+ *
+ * @author dimcho.nedev
+ *
+ */
+public class FtpFileTransferService implements Runnable {
+    private static final Logger LOGGER = Logger.getLogger(AbstractWrapDevice.class.getCanonicalName());
+
+    private File queueOfFileNames;
+
+    private List<String> fileNameRows;
+
+    private FtpConnectionManager connectionManager;
+
+    private FileRecycler fileRecycler;
+
+    public FtpFileTransferService(String pendingTransfersFileName,
+            FtpConnectionManager connectionManager,
+            FileRecycler fileRecycler)
+        throws IOException {
+        this.queueOfFileNames = new File(pendingTransfersFileName);
+
+        if (!queueOfFileNames.exists()) {
+            queueOfFileNames.createNewFile();
+        }
+
+        this.fileNameRows = new ArrayList<>();
+        this.connectionManager = connectionManager;
+        this.fileRecycler = fileRecycler;
+    }
+
+    @Override
+    public void run() {
+        try {
+            if (connectionManager.isAvailableForTransfer() && !isEmptyQueue()) {
+                // retrieves and removes the head of this queue
+                String filePath = pollNextLine();
+                File fileToTransfer = new File(filePath);
+                boolean isSuccessfulTransfered = connectionManager.transferData(fileToTransfer,
+                                                                                fileToTransfer.getName());
+                // TODO: Find a way to transfer an error message (if occurs) to the client
+
+                if (isSuccessfulTransfered) {
+                    synchronized (fileRecycler) {
+                        fileRecycler.addFile(filePath);
+                    }
+                }
+            }
+        } catch (IOException e) {
+            LOGGER.error(String.format("Failed to read from the \"%s\" file", queueOfFileNames.getName()), e);
+        }
+    }
+
+    /**
+     * Add a transfer task to a text file.
+     *
+     * @param pathToFile
+     *        - path to the file that will be transfered
+     * @throws IOException
+     *         thrown when failed to read/write from/to the queue file
+     */
+    public synchronized void addTransferTask(String pathToFile) throws IOException {
+        fileNameRows = FileUtils.readLines(queueOfFileNames);
+        fileNameRows.add(pathToFile);
+        FileUtils.writeLines(queueOfFileNames, fileNameRows);
+    }
+
+    private synchronized String pollNextLine() throws IOException {
```

Comment:
In fact this method is getting the file you need to transfer. Please use other name for it.